File: ChessProject-Java/src/main/java/com/solarwindsmsp/chess/PieceFactory.java
package com.solarwindsmsp.chess;

public class PieceFactory {

    public PieceInterface create(String type, PieceColor color) {

        if (type == null) {
            return null;
        }
        if (type.equalsIgnoreCase("Pawn")) {
            Pawn pawn = new Pawn(color);
            pawn.setCoordinates(new Coordinates(-1, -1));

            return pawn;
        }

        return null;
    }
}

File: ChessProject-PHP/src/Factory/PieceFactory.php
<?php

namespace SolarWinds\Chess\Factory;

use SolarWinds\Chess\Pawn;
use SolarWinds\Chess\Piece;
use SolarWinds\Chess\PieceColor;

class PieceFactory
{
    const PAWN = 1;

    public static function create(int $type, PieceColor $color): Piece
    {
        switch ($type){
            case self::PAWN:
                return new Pawn($color);
                break;
            default:
                throw new \Exception('Invalid type');
                break;
        }
    }
}